Domain.Sql.Tests/ScheduleCommandsDiagnosticTests.cs

```diff
@@ -0,0 +1,191 @@
+            Command<Order>.AuthorizeDefault = (order, command) => true;
+        }
+
+        private void TriggerExceptionOnOrderCommands()
+        {
+            orderRepository.GetEventStoreContext = () => { throw new Exception(); };
+        }
+
+        private void StopTriggeringExceptions()
+        {
+            orderRepository.GetEventStoreContext = () => new EventStoreDbContext();
+        }
+
+        private void ClearAbandonedAndPendingScheduledCommands()
+        {
+            Debug.WriteLine("Clearing�abandoned�and�pending�scheduled�commands.");
+            using (var db = new CommandSchedulerDbContext())
+            {
+                var commands = db.ScheduledCommands.Where(s => s.AppliedTime == null).ToList();
+                commands.ForEach(
+                                 s =>
+                                 {
+                                     s.AppliedTime = s.DueTime ?? Clock.Now();
+                                     s.FinalAttemptTime = null;
+                                 });
+
+                db.SaveChanges();
+            }
+        }
+
+        private async Task TriggerAbandonedCommandByException()
+        {
+            var order = CommandSchedulingTests.CreateOrder();
+            await accountRepository.Save(new CustomerAccount(order.CustomerId).Apply(new ChangeEmailAddress(Any.Email())));
+            await orderRepository.Save(order);
+            TriggerExceptionOnOrderCommands();
+
+            var orderScheduler = Configuration.Current.CommandScheduler<Order>();
+            await orderScheduler
+                .Schedule(new CommandScheduled<Order>
+                          {
+                              AggregateId = order.Id,
+                              Command = new Cancel()
+                          });
+
+            for (var i = 1; i < 10; i++)
+            {
+                await AdvanceDefaultClock(TimeSpan.FromHours(1));
+            }
+
+            StopTriggeringExceptions();
+        }
+
+        private async Task TriggerOverdueCommandByException()
+        {
+            StopTriggeringExceptions();
+            var order = CommandSchedulingTests.CreateOrder();
+            await accountRepository.Save(new CustomerAccount(order.CustomerId).Apply(new ChangeEmailAddress(Any.Email())));
+            await orderRepository.Save(order);
+            TriggerExceptionOnOrderCommands();
+
+            var orderScheduler = Configuration.Current.CommandScheduler<Order>();
+            await orderScheduler
+                .Schedule(new CommandScheduled<Order>
+                          {
+                              AggregateId = order.Id,
+                              Command = new Cancel()
+                          });
+
+            await AdvanceDefaultClock(TimeSpan.FromHours(25));
+
+            StopTriggeringExceptions();
+        }
+
+        private async Task AdvanceDefaultClock(TimeSpan timeSpan)
+        {
```

Comment:
Since this looks like it's just for debugging purposes while writing the tests, it can probably be deleted.